sycl/doc/extensions/experimental/sycl_ext_oneapi_bindless_images/sycl_ext_oneapi_bindless_images.asciidoc

```diff
@@ -0,0 +1,1826 @@
+  image_channel_order channel_order;
+  image_type type;
+  unsigned int num_levels;
+
+  image_descriptor(sycl::range<1> dims, sycl::image_channel_order channel_order,
+                   sycl::image_channel_type channel_type, 
+                   image_type type = image_type::standard, 
+                   unsigned int num_levels = 1);
+
+  image_descriptor(sycl::range<2> dims, sycl::image_channel_order channel_order,
+                   sycl::image_channel_type channel_type, 
+                   image_type type = image_type::standard, 
+                   unsigned int num_levels = 1);
+
+  image_descriptor(sycl::range<3> dims, sycl::image_channel_order channel_order,
+                   sycl::image_channel_type channel_type, 
+                   image_type type = image_type::standard, 
+                   unsigned int num_levels = 1);
+};
+
+}
+```
+
+The image descriptor represents the image dimensions, channel type, and channel 
+order. A `type` member is also present to allow for implementation of mipmapped 
+and interop images.
+
+The `image_descriptor` shall be default constructible and follow by-value 
+semantics.
+
+[NOTE]
+====
+Additional future `image_type`s _may_ include "layered" and/or "cubemap".
+====
+
+Note that `image_channel_type` and `image_channel_order` existed in SYCL 1.2.1,
+but were removed in SYCL 2020 in favor of a single, unified enum class.
+We propose separating them again to enable better flexibility
+and to avoid combinatorial complexity.
+
+For the `standard` image type, the value of `num_levels` must be `1`.
+
+Only mipmap image types support more than one level.
+
+The `type` member will inform the implementation of the type of image to 
+create, allocate, or free.
+
+
+=== Allocating image memory
+
+The process of creating an image is two-fold:
+allocate an image's memory, then create an image handle from the allocation.
+Allocation of image memory can be achieved in two ways. 
+
+==== Allocating non-USM image memory 
+
+```cpp
+namespace sycl::ext::oneapi::experimental {
+
+struct image_mem_handle {
+  using raw_handle_type = /* implementation defined */;
+  raw_handle_type raw_handle;
+}
+
+class image_mem {
+
+  image_mem();
+  image_mem(const image_mem &) = delete; // no copy-construct
+
+  image_mem(image_mem &&rhs) noexcept; // move-constructor is allowed
+  image_mem(const sycl::context &syclContext, const image_descriptor &desc);
+
+  ~image_mem();
+
+  image_mem &operator=(image_mem &&); // move-assignment is allowed
+  image_mem &operator=(image_mem &) = delete; // no copy-assignment
+
+  image_mem_handle get_handle() const;
+  image_descriptor get_descriptor() const;
+  sycl::context get_context() const;
+
+  sycl::range<3> get_range() const;
+  sycl::image_channel_type get_image_channel_type() const;
+  sycl::image_channel_type get_image_channel_order() const;
+  unsigned int get_image_num_channels() const;
+  image_type get_type() const;
+
+  image_mem_handle get_mip_level() const;
```

Comment:
```suggestion
  unsigned int get_mip_level() const;
```

I assume?

The `image_descriptor` APIs call this "num_levels".  Use consistent terminology.  If this is specific to mipmaps, maybe "mip_levels" or "mipmap_levels" is a good name to use everywhere.

Reply:
`get_mip_level` returns a handle to the image memory of a specific level in the mipmap, and not the number of mipmap levels. This enables users to perform operations on the memory of individual images in the mipmap.

I can see how the name might cause some confusion though. Perhaps we should rename to `get_mip_level_mem_handle` or similar.